### PR TITLE
feat(marketing): blog site

### PR DESCRIPTION
**Pull request by james-a-morris**
Introduces individual blog pages - it's recommended to use `/blog/tim` to find an example of a specific blog page.

### CODE REVIEW

File: src/app/_lib/contentful.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  entrySlugId: string,
+  tags: string[],
+  limit: number,
+): Promise<BlogPostType[]> {
+  const client = getProductionClient();
+  const options = {
+    content_type: contentType,
+    limit,
+    "fields.content[exists]": true, // no empty posts
+    "fields.tag[in]": tags.join(","), // get posts with same tags
```
Comment (gsteenkamp89) [inline src/app/_lib/contentful.ts:94]: be careful here: this is **case sensitive**.  So, for example, the string `uma` will not match any entries if the tag is `UMA` on contentful.

Comment (james-a-morris) [reply src/app/_lib/contentful.ts:94]: Great callout +1

**Review by gsteenkamp89 — APPROVED** [review]
From what I can tell this is a very solid start! 👏

File: src/app/(routes)/blog/[slug]/_components/article-content.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    const data = node.data.target as Asset<"WITHOUT_UNRESOLVABLE_LINKS", string>;
+    const { file, description, title } = data.fields;
+    const mimeGroup = file?.contentType.split("/")[0]; // image / video etc
+    switch (mimeGroup) {
+      case "image":
+        return <ContentfulImage image={data} />;
+      // TODO: test this, make custom component if necessary
+      case "video":
+        return (
+          <video title={title} aria-description={description} src={file?.url}>
```
Comment (gsteenkamp89) [inline src/app/(routes)/blog/[slug]/_components/article-content.tsx:124]: I think the `src` here also needs the protocol, same as in`<ContentfulImage />`.
```ts
const urlWithProtocol = `https:${file?.url}`;
```

File: src/app/(routes)/blog/[slug]/_components/background-banner.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { twMerge } from "@/app/_lib/tw-merge";
+import Image from "next/image";
+import backgroundBanner from "@/app/_assets/blog-background.png";
+
+export default function BackgroundBanner({ offsetTop }: { offsetTop: number }) {
+  return (
+    <Image
+      className={twMerge("absolute left-1/3 z-0 hidden sm:block", `top-[${offsetTop}px]`)}
```
[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/app/(routes)/blog/[slug]/_components/background-banner.tsx:8]: Yep! This is specifically scoped variable meant to hold pixel values

File: src/app/(routes)/blog/[slug]/_components/article-content.tsx (hunk 2 above)
Comment (james-a-morris) [reply src/app/(routes)/blog/[slug]/_components/article-content.tsx]: https://github.com/across-protocol/across.to/pull/56/commits/5f11c09dbff6ea8520f7f4de6946926715136f19

File: src/app/_lib/contentful.ts (hunk 1 above)
Comment (james-a-morris) [reply src/app/_lib/contentful.ts]: https://github.com/across-protocol/across.to/pull/56/commits/0cdc9664dfd3f35884e5f3890b76030b351c5727

File: src/app/(routes)/blog/[slug]/_components/background-banner.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dohaki) [reply src/app/(routes)/blog/[slug]/_components/background-banner.tsx:8]: Hmm, you sure that this works? I always thought Tailwind's build process can not handle these type of dynamic class names... See https://tailwindcss.com/docs/content-configuration#dynamic-class-names

**Review by dohaki — APPROVED** [review]
Great stuff 💪 Left one small remark

File: src/app/(routes)/blog/[slug]/_components/background-banner.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/app/(routes)/blog/[slug]/_components/background-banner.tsx]: Made this static - resolving https://github.com/across-protocol/across.to/pull/56/commits/baae53f1efa1e69537d8d2b4d77d0213c01a3fc4

**Pull request merged**